src/business_logic/transaction/internal_objects.rs

```diff
@@ -166,3 +179,376 @@ impl InternalDeploy {
+    }
+
+    pub fn verify_version(&self) -> Result<(), TransactionError> {
+        // no need to check if its lesser than 0 because it is an usize
+        if self.version > u64::pow(2, 63) {
+            return Err(TransactionError::StarknetError(
+                "The sender_address field in Declare transactions of version 0, sender should be 1"
+                    .to_string(),
+            ));
+        }
+
+        if !self.max_fee.is_zero() {
+            return Err(TransactionError::StarknetError(
+                "The max_fee field in Declare transactions of version 0 must be 0.".to_string(),
+            ));
+        }
+
+        if !self.nonce.is_zero() {
+            return Err(TransactionError::StarknetError(
+                "The nonce field in Declare transactions of version 0 must be 0.".to_string(),
+            ));
+        }
+
+        if !self.signature.len().is_zero() {
+            return Err(TransactionError::StarknetError(
+                "The signature field in Declare transactions must be an empty list.".to_string(),
+            ));
+        }
+        Ok(())
+    }
+
+    pub fn apply_specific_concurrent_changes(
+        &self,
+        state: &mut CachedState<InMemoryStateReader>,
+        general_config: StarknetGeneralConfig,
+    ) -> Result<TransactionExecutionInfo, TransactionError> {
+        self.verify_version()?;
+        // validate transaction
+        let mut resources_manager = ExecutionResourcesManager::default();
+
+        let validate_info =
+            self.run_validate_entrypoint(state, &mut resources_manager, general_config)?;
+
+        let updates_tracker_state = UpdatesTrackerState::new(state.clone());
+        let actual_resources = calculate_tx_resources(
+            resources_manager,
+            &vec![validate_info.clone()],
+            TransactionType::Declare,
+            updates_tracker_state,
+            None,
+        )?;
+
+        Ok(
+            TransactionExecutionInfo::create_concurrent_stage_execution_info(
+                validate_info,
+                None,
+                actual_resources,
+                Some(self.tx_type.clone()),
+            ),
+        )
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Internal Account Functions
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    pub fn get_execution_context(&self, n_steps: u64) -> TransactionExecutionContext {
+        TransactionExecutionContext::new(
+            self.sender_address.clone(),
+            self.hash_value.clone(),
+            self.signature.clone(),
+            self.max_fee,
+            self.nonce.clone(),
+            n_steps,
+            self.version,
+        )
+    }
+
+    pub fn run_validate_entrypoint(
+        &self,
+        state: &mut CachedState<InMemoryStateReader>,
+        resources_manager: &mut ExecutionResourcesManager,
+        general_config: StarknetGeneralConfig,
+    ) -> Result<Option<CallInfo>, ExecutionError> {
+        if self.version > u64::pow(2, 63) {
```

Comment:
Same as the other comment with `pow(2, 63)`.

Reply:
done

Reply:
Are you sure it's done?

Reply:
i didnt upload the changes yet, i was checking something